src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -86,244 +128,146 @@ class Ignored(Exception):
-                raise utils.WrongConfiguration(
-                    "Missing token specification. Either configure a static "
-                    "token or the path to the token file."
-                )
 
-        try:
-            self.ignore_regex = self.config["ignore_regex"]
-            self.log.debug("Ignoring files matching '%s'.", self.ignore_regex)
-        except KeyError:
-            pass
+    def get_data_type(self):
+        return "metadata"
 
-        try:
-            self.file_start_index = self.config["files_in_scan_start_index"]
-        except KeyError:
-            self.file_start_index = 0
+    def process(self, local_path, metadata, data=None):
+        """Send the file to the ASAP::O producer
 
-        self.lock = threading.Lock()
+        Asapo data_source and index are chosen by the incoming files in format
+        (<scan_id, <file_id>) -> (stream, id)
 
-    def _check_config(self):
-        failed = False
-
-        for i in self.required_parameter:
-            if i not in self.config:
-                self.log.error(
-                    "Wrong configuration. Missing parameter: '%s'", i
-                )
-                failed = True
-
-        if failed:
-            raise utils.WrongConfiguration(
-                "The configuration has missing or wrong parameters."
-            )
-
-    def _set_ingest_mode(self, mode):
-        if mode == "INGEST_MODE_TRANSFER_METADATA_ONLY":
-            self.ingest_mode = (asapo_producer
-                                .INGEST_MODE_TRANSFER_METADATA_ONLY)
-            self.data_type = "metadata"
-        # TODO data forwarding
-#        elif mode == "INGEST_MODE_TRANSFER_DATA":
-#            self.ingest_mode = asapo_producer.INGEST_MODE_TRANSFER_DATA
-#            self.data_type = "data"
-#        elif mode == "DEFAULT_INGEST_MODE":
-#            self.ingest_mode = asapo_producer.DEFAULT_INGEST_DATA
-#            self.data_type = "data"
-        else:
-            raise utils.NotSupported("Ingest mode '{}' is not supported"
-                                     .format(mode))
+        Args:
+            local_path: The absolute path where the file was written
+            metadata: The metadata to send as dict
+            data (optional): the data to send
+        """
 
-    def get_data_type(self):
-        return self.data_type
-
-    def _create_producer(self, stream):
-        token = self.token or self._get_token()
-        beamtime = self.beamtime or self._get_beamtime()
-        self.log.debug("type of beamtime%s", type(beamtime))
-
-        config = dict(
-            endpoint=self.endpoint,
-            beamtime_id=beamtime,
-            beamline=self.beamline,
-            stream=stream,
-            token=token,
-            nthreads=self.n_threads,
-            timeout_sec=self.timeout
-        )
+        if self._config_is_modified() or self.asapo_worker is None:
+            self.config.update(utils.load_config(self.config["user_config_path"]))
```

Comment:
For testing, this is good, but for production the user should not be able to overwrite all settings (can be changed in a later PR).